### DISCUSSION TITLE
[Q&A] How to Streamlink API Get DashStream Video and Audio separately  ?

### DISCUSSION THREAD

**Discussion by DevataDev** [streamlink/streamlink#5339] [Q&A] [answered]
### Description

Hi all,
Is it possible to get the DashStream Audio and Video Separately rather than using open() that will call default FfmpegMuxer ?

Thank you

**Comment by bastimeyer** [comment, marked as answer]
No, not directly. You'd have to create a `DASHStreamReader` instance yourself with the `DASHStream`'s correct `Representation` object (`DASHStream.video_representation` / `DASHStream.audio_representation`), similar to what `DashStream.open()` does:
https://github.com/streamlink/streamlink/blob/5.5.1/src/streamlink/stream/dash.py#L371-L392

The reason for that is that `DASHStream.parse_manifest()` returns `DashStream` instances only from the cartesian product of all available video and audio representations of the selected DASH period:
https://github.com/streamlink/streamlink/blob/5.5.1/src/streamlink/stream/dash.py#L332

In order for it to always return `DASHStream` instances with only a video representation or only an audio representation, the respective video and audio lists would need to be initialized with a `None` item, but it doesn't do that, as you can see here:
https://github.com/streamlink/streamlink/blob/5.5.1/src/streamlink/stream/dash.py#L286-L304

We could add a parameter (or two) to `DASHStream.parse_manifest()`, so that it also returns video-only and audio-only streams, or we could just let it do that by default. The problem I see with doing it by default though is that it generates lots of potentially unnecessary and confusion streams, and that's a problem because of lock of a proper stream collection/selection implementation described in #4902.

**Reply by DevataDev** [reply]
Hi @bastimeyer 
Thanks for the guidance and hint :) Cheers :)
